iOS/Flows/Conversation/Cells/MessageCell.swift

```diff
@@ -123,6 +123,49 @@ class MessageCell: UICollectionViewCell {
         self.handleDetailVisibility(areDetailsFullyVisible: areDetailsFullyVisible)
     }
 
+    private var messageController: MessageController?
+    private var messageSubscriptions: Set<AnyCancellable> = []
+    private var messageTasks = TaskPool()
+
+    private func subscribeToUpdatesIfNeeded(for messageable: Messageable) {
+        // If we're already subscribed to message updates, don't do it again.
+        guard messageable.id != self.messageController?.messageId else { return }
+
+        self.messageController = ChatClient.shared.messageController(for: messageable)
+
+        self.messageController?.reactionsPublisher
+            .mainSink(receiveValue: { [unowned self] _ in
+                Task {
+                    await self.refreshFooter()
+                }.add(to: self.messageTasks)
+            }).store(in: &self.messageSubscriptions)
+
+        self.messageController?.repliesChangesPublisher
+            .mainSink(receiveValue: { [unowned self] _ in
+                Task {
+                    await self.refreshFooter()
+                }.add(to: self.messageTasks)
+            }).store(in: &self.messageSubscriptions)
+    }
+
+    /// Gets the latest state of the message and updates the footer with that new state.
+    private func refreshFooter() async {
+        try? await self.messageController?.synchronize()
```

Comment:
Calling synchronize triggers a GetMessage call. We now only call it when a reaction or replies event occurs to cut down on unneeded calls.